src/fides/api/util/connection_type.py

```diff
@@ -94,38 +94,9 @@ def saas_request_type_filter(connection_type: str) -> bool:
         if template is None:  # shouldn't happen, but we can be safe
             return False
 
-        saas_config = SaaSConfig(**yaml.safe_load(template.config).get("saas_config"))
-        has_access = bool(
-            next(
-                (
-                    request.read
-                    for request in [
-                        endpoint.requests for endpoint in saas_config.endpoints
-                    ]
-                ),
-                None,
-            )
-        )
-        has_erasure = (
-            bool(
-                next(
-                    (
-                        request.update or request.delete
-                        for request in [
-                            endpoint.requests for endpoint in saas_config.endpoints
-                        ]
-                    ),
-                    None,
-                )
-            )
-            or saas_config.data_protection_request
-        )
-        has_consent = saas_config.consent_requests
-
-        return bool(
-            (ActionType.consent in action_types and has_consent)
-            or (ActionType.access in action_types and has_access)
-            or (ActionType.erasure in action_types and has_erasure)
```

Comment:
This logic was moved to the `SaaSConfig` https://github.com/ethyca/fides/pull/4374/files#diff-e02882cee5e6ee2325e4d613abed561b386e3e15064d0a1cbb41785a5d3f6d1aR483

Reply:
nice! like the refactor here 👍